TTbarAnalysis.py

```diff
@@ -713,117 +505,133 @@ def definePlots(self, t, noSel, sample=None, sampleCfg=None):
+                                    if self.doYields:
+                                        yield_object.addYields(sel, f"{channel}_{nbrj}bJets_{tagger}{wp}","2Lep(OS) + {nbrj} bJets {tagger}{wp} + {metcut} (channel : {optstex[channel]})")
+
+                ## two-tag count method
+                sel2l2j_METCut = sel2l2j.refine(f"{nbrj}Jets{channel}selection_METcut", cut=[ corrMET.pt < 80.])
                 if make_TwoTagCountPlots:
-                    # to avoid refining the same selection in loops of ptBin
-                    TwoLeptonsTwoJets_METCut = TwoLeptonsTwoJets.refine("{0}Jets{1}selection_METcut".format(nbrj, channel), cut=[ corrMET.pt < 80.])
-                    plots +=(TwoTagCount(channel, jets, TwoLeptonsTwoJets_METCut, nbrj, era, isMC))
+                    for iPT in range(-1, len(lowerPtBinEdges)):
+                        ptBin = getPtLabel(iPT)
+                        if ptBin == "Inclusive":
+                            sel_twotag_pt = sel2l2j_METCut
+                        else:
+                            if ptBin =="600toInf":
+                                minpt = 600.
+                                maxpt = 1000.
+                            else:
+                                minpt = lowerPtBinEdges[iPT]
+                                maxpt = lowerPtBinEdges[iPT+1]
+                            sel_twotag_pt = sel2l2j_METCut.refine(f"{sel2l2j_METCut.name}_2Jets_{ptBin}ptBin_cut{nbrj}", cut=[ op.in_range(minpt, jets[0].pt, maxpt), op.in_range(minpt, jets[1].pt, maxpt) ])
+                        if isMC:
+                            nTrueB = op.rng_count(jets[:2], lambda j : j.hadronFlavour == 5)
+                        else:
+                            nTrueB = op.c_int(0)
+                        sel_twotag_pt_twoTrueB = sel_twotag_pt.refine(f"{sel_twotag_pt.name}_twoTrueB", cut=(nTrueB == 2))
+                        for tagger,workingPoints in bTaggingWPs_era[era].items():
+                            for wp,thresh in workingPoints.items():
+                                key = f"{tagger}{wp}"
+                                nTaggedB = op.rng_count(jets[:2], partial((lambda j,fun=None,val=None : fun(j) >= val), fun=t_bDisc[tagger], val=thresh))
+                                ## now two plots: two true b-jets -> n pass; two tagged: n true
+                                sel_twotag_pt_twoTaggedB = sel_twotag_pt.refine(f"{sel_twotag_pt.name}_twoTaggedB_{key}", cut=(op.rng_len(bjets_pass) == 2))
+                                plots += [
+                                    Plot.make1D(f"{channel}_Events_2l{nbrj}j_nbtagged_TruthFlav_2Beauty_{key}_{ptBin}",
+                                        nTaggedB, sel_twotag_pt_twoTrueB,
+                                        EqBin(3, 0., 3.), title="N b-tags for events with two true b's",
+                                        plotopts=utils.getOpts(channel, **{"log-y": True})),
```

Comment:
I think here nTagged should return ; 0 btagged jet, 1btagged jet, **OR** 2btagged jets 
the selection requires 2B truth Flavour, respectively these events will be in bin 0, 1 and 2  => very close to the plot that we need,  take a look at the draft attached below. 
But you didn't take into account data in nTagged: it can be in an extra bin out of the of mc bins or in bin2 if you want since we are looking for data 2btagged. 
![OldPlot](https://user-images.githubusercontent.com/47111455/87227963-7269b200-c39e-11ea-9882-7b95c4835bdf.jpeg)